Content.Shared/Construction/SharedFlatpackSystem.cs

```diff
@@ -147,4 +168,16 @@ public void SetupFlatpack(Entity<FlatpackComponent?> ent, Entity<MachineBoardCom
 
         return cost;
     }
+
+    public Dictionary<string, int> GetFlatpackCreationCostForComputer(Entity<FlatpackCreatorComponent> entity)
```

Comment:
there's a lot of duplication in the code around the fact that you have to do TryComps in order to call either this method or the one for machine boards. Make a generic one that only takes an entityuid and does the trycomps inside so you can reduce duplication, namely in the UI.